r_basics_visualize_data/r_basics_visualize_data.md

```diff
@@ -0,0 +1,1064 @@
+
+Then, work through the exercises in the `visualize_exercises.Rmd` file, adding and updating code as indicated.
+
+Stop when it says "Stop here".  We'll go over the solutions in the next section.
+
+### Solutions
+
+If you like, you can open the solutions version of `visualize_exercises.Rmd` by using your file browser to open `visualize_solutions.Rmd`.
+
+You had three tasks to complete.  We'll go through them one at a time.
+
+In your first task, you were asked to do what we already did once in this module: create a histogram of Covid tests as a function of `pan_day`.  There were three blanks to fill in, and three numbered instructions.
+
+![Lines 18-31 of visualize_exercises.Rmd](media/task_1.png)<!-- style = "max-width:700px; border: 1px solid rgb(var(--color-highlight))" -->
+
+Here's the solution that you should place in that code chunk:
+
+``` r
+ggplot(data = covid_testing) +
+  geom_histogram(mapping = aes(x = pan_day))
+```
+
+
+In the second task, you were asked to rework that code with a bin width ("binwidth" in code) that corresponded to daily counts:
+
+![Lines 44-50 of visualize_exercises.Rmd](media/task_2.png)<!-- style = "max-width:700px; border: 1px solid rgb(var(--color-highlight))" -->
+
+Here's the solution that you should place in that code chunk:
+
+``` r
+ggplot(data = covid_testing) +
+  geom_histogram(mapping = aes(x = pan_day), binwidth = 1)
+```
+
+Finally, for the third task, you had to create a new code chunk by using copy and paste, **and** you had to add an aesthetic mapping using "fill".  Complicated!
+
+![Lines 58-60 of visualize_exercises.Rmd](media/task_3.png)<!-- style = "max-width:700px; border: 1px solid rgb(var(--color-highlight))" -->
+
+
+And here's the solution:
+
+```
+ggplot(data = covid_testing) +
+  geom_histogram(mapping = aes(x = pan_day, fill = result), binwidth = 1)
+```
+
+### Setting vs. Mapping Aesthetics
+
+In the third task you just completed (if you didn't actually do it, at least go back one section and read about the three tasks), you **mapped** the fill aesthetic to the **result** variable, by writing "fill = result" inside the **aes()** function:
+
+<div style = "align-items: center; display: flex;">
+
+<div style = "margin: 1rem; max-width: 45%; float:left;"> ![Histogram of Covid tests as a function of `pan_day`.  The bars forming the histogram have three colors: a small blue section at the bottom, representing positive results, a much larger green section in the middle for negative results, and a tiny red section at the top for invalid test results.](media/multicolor_histogram.png)
+</div>
+<div style = "margin: 1rem auto; max-width: 45%; float:left;">
+
+<pre>
+ggplot(data = covid_testing) +
```

Comment:
probably needs to be wrapped -- currently extends straight off the screen requiring scrolling to view entire line

Reply:
stupid liascript rendering ugh.  Fixing.